plugin/pkg/scheduler/extender.go

```diff
@@ -82,53 +83,89 @@ func NewHTTPExtender(config *schedulerapi.ExtenderConfig, apiVersion string) (al
-		prioritizeVerb: config.PrioritizeVerb,
-		weight:         config.Weight,
-		client:         client,
+		extenderURL:      config.URLPrefix,
+		filterVerb:       config.FilterVerb,
+		prioritizeVerb:   config.PrioritizeVerb,
+		weight:           config.Weight,
+		client:           client,
+		nodeCacheCapable: config.NodeCacheCapable,
 	}, nil
 }
 
 // Filter based on extender implemented predicate functions. The filtered list is
 // expected to be a subset of the supplied list. failedNodesMap optionally contains
 // the list of failed nodes and failure reasons.
-func (h *HTTPExtender) Filter(pod *v1.Pod, nodes []*v1.Node) ([]*v1.Node, schedulerapi.FailedNodesMap, error) {
-	var result schedulerapi.ExtenderFilterResult
+func (h *HTTPExtender) Filter(pod *v1.Pod, nodes []*v1.Node, nodeNameToInfo map[string]*schedulercache.NodeInfo) ([]*v1.Node, schedulerapi.FailedNodesMap, schedulerapi.Annotations, error) {
+	var (
+		result         schedulerapi.ExtenderFilterResult
+		nodeList       *v1.NodeList
+		nodeNames      *[]string
+		nodeResult     []*v1.Node
+		podAnnotations schedulerapi.Annotations
+		args           *schedulerapi.ExtenderArgs
+	)
 
 	if h.filterVerb == "" {
-		return nodes, schedulerapi.FailedNodesMap{}, nil
+		return nodes, schedulerapi.FailedNodesMap{}, nil, nil
 	}
 
-	nodeItems := make([]v1.Node, 0, len(nodes))
-	for _, node := range nodes {
-		nodeItems = append(nodeItems, *node)
+	if h.nodeCacheCapable {
+		nodeNameSlice := make([]string, 0, len(nodes))
+		for _, node := range nodes {
+			nodeNameSlice = append(nodeNameSlice, node.Name)
+		}
+		nodeNames = &nodeNameSlice
+	} else {
+		nodeList = &v1.NodeList{}
+		for _, node := range nodes {
+			nodeList.Items = append(nodeList.Items, *node)
+		}
 	}
-	args := schedulerapi.ExtenderArgs{
-		Pod:   *pod,
-		Nodes: v1.NodeList{Items: nodeItems},
+
+	args = &schedulerapi.ExtenderArgs{
+		Pod:       *pod,
+		Nodes:     nodeList,
+		NodeNames: nodeNames,
 	}
 
-	if err := h.send(h.filterVerb, &args, &result); err != nil {
-		return nil, nil, err
+	if err := h.send(h.filterVerb, args, &result); err != nil {
+		return nil, nil, nil, err
 	}
 	if result.Error != "" {
-		return nil, nil, fmt.Errorf(result.Error)
+		return nil, nil, nil, fmt.Errorf(result.Error)
 	}
 
-	nodeResult := make([]*v1.Node, 0, len(result.Nodes.Items))
-	for i := range result.Nodes.Items {
-		nodeResult = append(nodeResult, &result.Nodes.Items[i])
+	if h.nodeCacheCapable && result.NodeNames != nil {
+		nodeResult = make([]*v1.Node, 0, len(*result.NodeNames))
+		for i := range *result.NodeNames {
+			nodeResult = append(nodeResult, nodeNameToInfo[(*result.NodeNames)[i]].Node())
+		}
+	} else if result.Nodes != nil {
+		nodeResult = make([]*v1.Node, 0, len(result.Nodes.Items))
+		for i := range result.Nodes.Items {
+			nodeResult = append(nodeResult, &result.Nodes.Items[i])
+		}
 	}
-	return nodeResult, result.FailedNodes, nil
+
+	if result.PodAnnotations != nil {
+		podAnnotations = *result.PodAnnotations
```

Comment:
same comment re: we are ridding annotations in 1.6... 

Are there other things you are passing using this mechanism?